osbs/utils/__init__.py

```diff
@@ -588,6 +590,13 @@ def retry(*args, **kwargs):
     return do_retry_on_exception
 
 
+def user_warning_log_handler(self, msg, *args, **kwargs):
+    if isinstance(msg, dict):
+        msg = json.dumps(msg, indent=2)
+
+    self._log(USER_WARNING_LEVEL, msg, args, **kwargs)
```

Comment:
We quite violates how logger works here. It expect formatting string and args are applied to it, but when dict is passed into it, it's transformed into string, can it have some side effects?

Reply:
Good catch :+1: